proxy.go

```diff
@@ -124,6 +125,33 @@ func newClient(proxy *proxy, conn net.Conn) *client {
 	}
 }
 
+func (proxy *proxy) restoreTokens(vm *vm, tokens []string) error {
```

Comment:
Shouldn't this be `tokens []Token`? Also, what if `vm == nil`?

Reply:
Unfortunately proxy.allocateTokens() returns []string. I'd like to avoid writing a conversion loop from []string to []Token in registerVM() and revert conversion loop in storeVMState().